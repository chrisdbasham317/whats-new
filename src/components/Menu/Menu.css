li {
  margin-bottom: 5%;
  text-align: center;
  width: 100%;
}

li:hover {
  background: #FEAE03;
}