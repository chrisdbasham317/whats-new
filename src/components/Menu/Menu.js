import React from 'react';
import './Menu.css'




const Menu = ({ text, dataSets, updateData }) => {
  return (
    <li
      className={`li li--${text}`}
      onClick={event => updateData(dataSets[text])}
    >{text.toUpperCase()}</li> 
  )
}

export default Menu;