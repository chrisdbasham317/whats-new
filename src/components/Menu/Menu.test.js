import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';
import local from '../../data/local'

describe('Menu', () => {
  let wrapper
  const mockUpdateData = jest.fn();
  
  beforeEach(() => {
    wrapper = shallow(<Menu
      key='local'
      text='local'
      dataSets={ { local: local } }
      updateData={mockUpdateData}
      />);
    });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should run updateData when button is clicked', () => {
    wrapper.find('li').simulate('click');
    expect(mockUpdateData).toHaveBeenCalledWith(local);
  });
})