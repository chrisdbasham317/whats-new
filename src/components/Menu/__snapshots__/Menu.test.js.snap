// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<li
  className="li li--local"
  onClick={[Function]}
>
  LOCAL
</li>
`;
